### PR TITLE
[tree-sitter], [tree-sitter-c], and [tree-sitter-json] New ports

### PR DESCRIPTION
**Pull request by SamuelMarks**
**Describe the pull request**
> Tree-sitter is a parser generator tool and an incremental parsing library. It can build a concrete syntax tree for a source file and efficiently update the syntax tree as the source file is edited.

https://github.com/tree-sitter/tree-sitter

This port is for the C grammar:
> C grammar for tree-sitter. Adapted from a C99 grammar.

https://github.com/tree-sitter/tree-sitter-c

This port is for JSON grammar:
> JSON grammar for [tree-sitter](https://github.com/tree-sitter/tree-sitter)

https://github.com/tree-sitter/tree-sitter-json

- #### What does your PR fix?
New ports

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
All

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**
Maybe wait until these PRs are addressed by the maintainers?
  - https://github.com/tree-sitter/tree-sitter/pull/1749
  - https://github.com/tree-sitter/tree-sitter-c/pull/102
  - https://github.com/tree-sitter/tree-sitter-json/pull/25

### CODE REVIEW

File: ports/tree-sitter-c/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,19 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO            "SamuelMarks/${PORT}"
```
Comment (FrankXie05) [inline ports/tree-sitter-c/portfile.cmake:3]:
```suggestion
    REPO            "tree-sitter/tree-sitter-c"
```
It's not an official repo

File: ports/tree-sitter-c/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    REF             bf97c9482e90af607ff76346e14d4626aba0f1f9
+    SHA512          779f2eb4e57cab5fc26ed272af8e2a0249f4a683c7ee1ca929bacec0f925cd923e2a339423c1975b1c2081639cb9326f6760c163c94af1e4fc5d776712614aa9
+    HEAD_REF        cmake
```
Comment (FrankXie05) [inline ports/tree-sitter-c/portfile.cmake:6]:
```suggestion
    HEAD_REF       master
```

File: ports/tree-sitter-c/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+)
+
+vcpkg_cmake_configure(
```
Comment (FrankXie05) [inline ports/tree-sitter-c/portfile.cmake:9]: The official {SOURCE_PATH} does not contain CMakeLists.txt.

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Comment by FrankXie05** [comment]
@SamuelMarks Thank you for submitting the PR, but this PR contains the addition of new ports `[tree-sitter-c]` and `[tree-sitter]`, so #24767  should be closed in order to avoid taking up too many CI resources. 
And the addition of new ports needs to meet the following points: 

New port checklist:

- Check the name against https://repology.org/ 
- Check the name against Bing/Google. (If the first results aren't C++ related, try "name C++") 
- Check the source code for optional find_packages 
- Check that the versioning scheme and license match what upstream says 
- Check that the source code comes from the upstream project's authoritative source. 
- Check that the generated usage is accurate.

**Comment by FrankXie05** [comment]
Pinging @SamuelMarks  for response. Is work still being done for this PR?

**Comment by SamuelMarks** [comment]
@maxbrunsfeld doesn't want it https://github.com/tree-sitter/tree-sitter/pull/1749

**Pull request closed without merging**